Graphql and dotnet parser

linearcombination opened the pull request.

linearcombination committed:
Most changes to get graphql and new parser use

linearcombination committed:
Get HEAD sha1 value into 'up' makefile rule

linearcombination committed:
Rest of changes for update

linearcombination committed:
Fix accidental edit to graphql query

linearcombination committed:
Fix updates to get to passing tests

linearcombination committed:
Filter out asset directories which do not represent bible book codes

linearcombination committed:
Use different data API URL

The data API URL I have been using is having intermittent issues and
returning no data, so trying this one which seems to be working more
consistently this afternoon.

linearcombination committed:
Attempt to diagnose issue with .net execution in github actions

linearcombination committed:
Attempt to address github actions CI peculiarities

linearcombination committed:
Give a little more info about Github Action problem

linearcombination committed:
Run all we can as non-root user

linearcombination committed:
Attempt getting Github Action environment to use .NET successfully

linearcombination committed:
Skip test

linearcombination commented:
This is not ready to merge yet. Need to configure .NET setup for Github Actions as it requires to handle .NET on its own....

PurpleGuitar commented:
Thank you for the update Lang!  I've started reviewing, but I'll keep in mind more changes are coming.

linearcombination commented:
From: Craig Oliver ***@***.***>
Subject: Re: [WycliffeAssociates/DOC] Graphql and dotnet parser (PR #216)
Date: Mon, 05 Aug 2024 08:20:14 -0700

Thank you Craig! Yes, more changes to come. There is a cool project
called 'act' which lets you run github actions locally and so I will
be using it test having Github Actions install .NET (which I learned
Github Actions requires) without having to push changes. Very useful
when you are experimenting, just thought I would point it out if you
haven't heard of it, I only just discovered it:
https://github.com/nektos/act

> Thank you for the update Lang!  I've started reviewing, but I'll keep in mind more changes are coming.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/WycliffeAssociates/DOC/pull/216#issuecomment-2269329412
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>

linearcombination committed:
Update github actions workflow

Github actions requires to manage .NET itself for its CI/CD
run.

linearcombination committed:
Fix issue with dotnet cli command not invoking correctly

Research indicates that setting subprocess's shell to False could
resolve the error we are getting wherein the dotnet command we issue
gets mangled.

linearcombination committed:
Use absolute path to dotnet cli

linearcombination committed:
Help Python subprocess fine assets

linearcombination committed:
Overcome intermittent issues with data API using cache

linearcombination committed:
Overcome issues with Github Action maintained dotnet cli

Tell Github Actions to install it to the same place that we install
for normal use in our Dockerfile, that way DOTNET_ROOT will point to
the same place in both Github Actions servers and in production.

linearcombination committed:
Further attempt to make Github Actions dotnet install work

This time use the same install directory of /usr/share/dotnet for the
dotnet directory in both Github Actions dotnet setup and in our normal
operating environment

linearcombination committed:
Another attempt to make Github Actions happy

This time using subprocess.run and handing it a list of command
components rather than a string and not using shell=True.

linearcombination committed:
Make dotnet restore into /app in Github Actions

Otherwise Github Actions restores the dotnet project into a private
runner directory and thus the dll cannot be found.

Now backend tests pass, but still need to get frontend working in
Github Actions.

linearcombination committed:
See if Github Actions will use the dotnet we install ourselves

Dotnet works fine and all tests pass locally and would on our
destination server, but Github Actions server/runner environnment has
unusual handling of .NET for (yet) unknown reasons.

linearcombination committed:
Get logging debug feedback on dotnet project layout

linearcombination committed:
Use absolute path to 'ls' command

linearcombination committed:
Provide Pythonic means to print out directory layout

linearcombination committed:
Print out less information, just show the dotnet project

linearcombination committed:
Add dotnet build for Github Actions sake

dll wasn't present on Github Actions server, so let's do a fresh build
to see if that makes the dll available

linearcombination committed:
Prefer subprocess.run over subprocess.call

And do not use shell=True, more secure.
